File: sourceC# - Visual Studio/Workspaces/ESHOP/EShop/Entities/ProductPictureEntity.cs
using EShop.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EShop.Entities
{
    public class ProductPictureEntity : BaseEntity
    {
        public Guid Id { get; set; }
        public String Path { get; set; }
        public Boolean IsDefault { get; set; }
        public Guid ProductId { get; set; }
        public ProductEntity ProductEntity { get; set; }

        public ProductPictureEntity():base() { }

        public ProductPictureEntity(ProductPicture ProductPicture, params object[] args) :base(ProductPicture)
        {
		    foreach(object arg in args)
			{
                if (arg is Product Product)
                    ProductEntity = new ProductEntity(Product);				
			}
        }
    }

    public class ProductPictureSearchEntity : FilterEntity
    {
        public Guid? Id { get; set; }
        public String Path { get; set; }
        public Boolean? IsDefault { get; set; }
        public Guid? ProductId { get; set; }
    }
}

File: sourceC# - Visual Studio/Workspaces/ESHOP/EShop/FrontEnd/src/app/Module/MProductPicture/ProductPicture.Entity.ts
import { IEntity } from "../IEntity.Entity";
import { ProductEntity } from "../MProduct/Product.Entity";

export class ProductPictureEntity extends IEntity {
  Id: number = 0;
  Path: string = null;
  ProductId: number = null;
  IsDefault: boolean = false;
  Cx: number = null;
  ProductEntity: ProductEntity = null;
  public constructor() {
    super();
  }

}
